pkg/app/piped/notifier/slack.go

```diff
@@ -151,6 +152,28 @@ func (s *slack) buildSlackMessage(event model.NotificationEvent, webURL string)
 			{"Started At", makeSlackDate(d.CreatedAt), true},
 		}
 	}
+	generateDeploymentEventDataForTriggerFailed := func(app *model.Application, hash, msg string) {
+		link = fmt.Sprintf("%s/applications/%s?project=%s", webURL, app.Id, app.ProjectId)
+		commitURL, err := git.MakeCommitURL(app.GitPath.Repo.Remote, hash)
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("failed to get the URL for the specified commit: %v", err))
+		}
+		if commitURL == "" {
+			fields = []slackField{
+				{"Project", truncateText(app.ProjectId, 8), true},
+				{"Application", makeSlackLink(app.Name, link), true},
+				{"Kind", strings.ToLower(app.Kind.String()), true},
+			}
+		} else {
+			fields = []slackField{
+				{"Project", truncateText(app.ProjectId, 8), true},
+				{"Application", makeSlackLink(app.Name, link), true},
+				{"Kind", strings.ToLower(app.Kind.String()), true},
+				{"Commit", makeSlackLink(truncateText(msg, 8), commitURL), true},
+			}
+		}
```

Comment:
Kind of like this could be thought better in terms of readability as well as consistency:
```suggestion
		fields = []slackField{
			{"Project", truncateText(app.ProjectId, 8), true},
			{"Application", makeSlackLink(app.Name, link), true},
			{"Kind", strings.ToLower(app.Kind.String()), true},
		}
		if commitURL != "" {
			fields = append(fields, slackField{"Commit", makeSlackLink(truncateText(msg, 8), commitURL), true})
		}
```